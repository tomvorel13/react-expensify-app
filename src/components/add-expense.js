import React from 'react';

const AddExpensePage = () => (
    <div>
        This is from my AddExpense component!
    </div>
);

export default AddExpensePage;