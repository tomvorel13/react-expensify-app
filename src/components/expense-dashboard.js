import React from "react";
import ExpenseList from "./expense-list";
import ExpenseListFilters from "./ExpenseListFilters";
import ExpensesSummary from "./expenses-summary";
import LoginPage from "./login-page";

const ExpenseDashboardPage = () => (
  <div>
    <ExpensesSummary />
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;
