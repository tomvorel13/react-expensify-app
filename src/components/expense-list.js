import React from "react";
import { connect } from "react-redux";
import ExpenseListItem from "./expense-list-item";
import selectExpenses from "../selectors/expenses";

export const ExpenseList = props => (
  <div className="container">
    <div className="list-header">
      <div className="show-for-mobile">Expenses</div>
      <div className="show-for-larger">Expense</div>
      <div className="show-for-larger">Amount</div>
    </div>
    <div className="list-body">
      {props.expenses.map(expense => {
        return <ExpenseListItem key={expense.id} {...expense} />;
      })}
    </div>
  </div>
);

const mapStateToProps = state => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseList);
