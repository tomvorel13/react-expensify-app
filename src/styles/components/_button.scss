.button {
  background: $blue;
  color: white;
  display: inline-block;
  border: none;
  font-weight: 300;
  font-size: $font-size-large;
  line-height: 1;
  padding: $s-size $m-size;
  text-decoration: none;
}

.button--link {
  background: none;
}

.button--secondary {
  background: #888;
}
