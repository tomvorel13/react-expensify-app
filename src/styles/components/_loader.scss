.loader {
  align-items: center;
  justify-content: center;
  height: 100vh;
  width: 100vw;
  display: flex;
}

.loader__image {
  height: 6rem;
  width: 6rem;
}
