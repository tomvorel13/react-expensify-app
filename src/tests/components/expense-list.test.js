import React from "react";
import { shallow } from "enzyme";
import { ExpenseList } from "../../components/expense-list";
import expenses from "../fixtures/expenses";

test("should render expense list with expenses", () => {
  const wrapper = shallow(<ExpenseList expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();
});

test("should render expenselist with empty message", () => {
  const wrapper = shallow(<ExpenseList expenses={[]} />);
  expect(wrapper).toMatchSnapshot();
});
