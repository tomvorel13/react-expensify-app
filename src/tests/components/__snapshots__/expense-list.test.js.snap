// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense list with expenses 1`] = `
<div
  className="container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      Expenses
    </div>
    <div
      className="show-for-larger"
    >
      Expense
    </div>
    <div
      className="show-for-larger"
    >
      Amount
    </div>
  </div>
  <div
    className="list-body"
  >
    <ExpenseListItem
      amount={195}
      createdAt={0}
      description="Gum"
      id="1"
      note=""
    />
    <ExpenseListItem
      amount={109500}
      createdAt={-345600000}
      description="Rent"
      id="2"
      note=""
    />
    <ExpenseListItem
      amount={4500}
      createdAt={345600000}
      description="Credit Card"
      id="3"
      note=""
    />
  </div>
</div>
`;

exports[`should render expenselist with empty message 1`] = `
<div
  className="container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      Expenses
    </div>
    <div
      className="show-for-larger"
    >
      Expense
    </div>
    <div
      className="show-for-larger"
    >
      Amount
    </div>
  </div>
  <div
    className="list-body"
  />
</div>
`;
